src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -35,38 +35,39 @@ class BankManager {
     std::mutex m_mutex;
 };
 
-ov::Tensor Bank::get(const LazyTensor& tensor, const std::string& device) {
+std::size_t Bank::registerLT(const LazyTensor& tensor, const std::string& device) {
     const std::string& device_for_alloc = m_alloc_device.empty() ? device : m_alloc_device;
 
     std::lock_guard<std::mutex> guard(m_mutex);
 
     auto& device_bank = m_device_banks[device_for_alloc];
-
-    std::unique_lock<std::mutex> dev_guard(device_bank.mutex);
-    auto iter_device = device_bank.storage.find(tensor);
-
-    if (iter_device != device_bank.storage.end() && iter_device->second) {
-        // Already allocated
-        // tensor (the key) may be coming from a 2nd (3rd, ...) model
-        // detach it here just in case
-        const_cast<LazyTensor&>(tensor).detach();
-        return iter_device->second;
+    if (device_bank.registered_tensors.find(tensor) == device_bank.registered_tensors.end()) {
+        auto uid = uid_count++;
+        device_bank.registered_tensors[tensor] = uid;
+        device_bank.storage[uid] = {tensor, ov::Tensor()};
+        return uid;
     }
-    dev_guard.unlock();
 
-    // Allocation and evaluation needed
-    return eval_and_alloc(tensor, device_bank, device_for_alloc);
+    return device_bank.registered_tensors[tensor];
 }
 
-void Bank::registerLT(const LazyTensor& tensor, const std::string& device) {
+ov::Tensor Bank::get(std::size_t uid, const std::string& device) {
     const std::string& device_for_alloc = m_alloc_device.empty() ? device : m_alloc_device;
 
     std::lock_guard<std::mutex> guard(m_mutex);
 
     auto& device_bank = m_device_banks[device_for_alloc];
-    if (device_bank.storage.find(tensor) == device_bank.storage.end()) {
-        device_bank.storage[tensor] = ov::Tensor();
-    }
+
+    std::unique_lock<std::mutex> dev_guard(device_bank.mutex);
+    auto iter_device = device_bank.storage.find(uid);
+
+    NPUW_ASSERT(iter_device != device_bank.storage.end() && iter_device->second.tensor &&
+                "Tensor should be registered and allocated first!");
```

Comment:
I believe it should only be registerd at this point. Previously, if the device bank didn't have a device tensor, evaluation happened. Did this contract change now?

Reply:
We had this discussion before during serialization review. I believe we agreed that we would drop the contract where `get()` could allocate anything unregistered